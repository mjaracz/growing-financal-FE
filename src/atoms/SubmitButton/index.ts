export * from './SubmitButton';
export * from './useStyle';
