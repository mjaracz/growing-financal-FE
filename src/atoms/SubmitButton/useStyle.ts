import { makeStyles, createStyles, Theme } from '@material-ui/core';

export const useStyle = makeStyles((theme: Theme) => 
  createStyles({
    button: {
      width: 180,
      height: 50,
      color: 'rgba(0, 0, 0, 0.87)',
      borderRadius: 50,
      fontWeight: 'bold',
      backgroundColor: theme.palette.primary.main,
      transition: '0.5s',
      '&:hover': {
        backgroundColor: theme.palette.secondary.main,
      }
    }
  })
) 