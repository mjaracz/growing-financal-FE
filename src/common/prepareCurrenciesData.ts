import { CurrenciesResponse } from 'molecules/currencySelect';
import { CurrencyRate } from 'molecules/currencySelect';

export function prepareCurrenciesData(currenciesResponse: CurrenciesResponse): CurrencyRate[] {
  const currenciesKeys: string[] = Object.keys(currenciesResponse.rates);
  const currenciesValues: number[] = Object.values(currenciesResponse.rates);
  const preparedCurrencies = [];
  for (let i = 0; i < currenciesKeys.length; i++) {
    const currencyRate = {}
    currencyRate[currenciesKeys[i]] = currenciesValues[i];
    preparedCurrencies.push(currencyRate)
  }
  return preparedCurrencies;
}