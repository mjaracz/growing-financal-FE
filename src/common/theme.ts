import { createMuiTheme } from '@material-ui/core';

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: 'hsl(193, 95%, 68%)',
      dark: 'hsl(220, 13%, 18%)'
    },
    secondary: {
      main: 'hsl(281, 100%, 80%)'
    },
    background: {
      default: 'hsl(220, 13%, 18%)'
    },
    text: {
      secondary: 'hsl(0, 0%, 100%)',
      hint: 'rgb(215, 146, 247)'
    }  
  }
})