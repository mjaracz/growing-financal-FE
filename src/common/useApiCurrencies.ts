export const useApiCurrencies = () => {
  const baseUrl = 'http://api.exchangeratesapi.io/v1/latest';
  const apiKey = '481b03536817ae9a3877e3026e237f51';

  const getLatestRate = async () => 
    fetch(baseUrl + `?access_key=${apiKey}`)
      .then(res => res.json())
      .then((data) => {
        if (data.statusCode >= 400) {
          throw new Error('User code error 4** status')
        }
        return data;
      })

  return {
    getLatestRate,
    baseUrl,
    apiKey,
  }
}
