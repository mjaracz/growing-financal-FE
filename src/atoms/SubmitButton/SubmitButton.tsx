import React from 'react';
import { Button } from '@material-ui/core';
import { useStyle } from './useStyle';

export interface Props {
  sendFormData: (formData: unknown) => void;
  label: string;
}

export const SubmitButton = (props: Props) => {
  const styles = useStyle();

  return (
    <Button
      color='primary'
      variant='contained'
      onClick={props.sendFormData}
      className={styles.button}
      
    >
      {props.label}
    </Button>
  )
}