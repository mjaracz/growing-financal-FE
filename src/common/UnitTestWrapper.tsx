import React, { FC } from 'react';
import { ThemeProvider } from '@material-ui/core';
import { theme } from './theme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from 'redux/store';

export const UnitTestWrapper: FC = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          {children}
        </Provider>
      </ThemeProvider>
    </BrowserRouter>
  )
}