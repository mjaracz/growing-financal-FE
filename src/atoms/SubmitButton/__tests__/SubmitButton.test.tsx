import React from 'react';
import { render } from '@testing-library/react';
import { SubmitButton } from '../SubmitButton';
import { UnitTestWrapper } from 'common/UnitTestWrapper';

describe('<SubmitButton />', () => {
  it('when pass correct props should match the snapshot', () => {
    const { baseElement } = render(
      <UnitTestWrapper>
        <SubmitButton 
          sendFormData={jest.fn(() => {})} 
          label='mock label' 
        />
      </UnitTestWrapper>
    );
    expect(baseElement).toMatchSnapshot();
  })
})