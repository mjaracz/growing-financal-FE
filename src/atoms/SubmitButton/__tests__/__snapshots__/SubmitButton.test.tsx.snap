// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> when pass correct props should match the snapshot 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        mock label
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
